app/models/user.rb

```diff
@@ -6,8 +6,19 @@ class User < ActiveRecord::Base
 
   has_many :lists
 
+  before_create do |doc|
```

Comment:
Not sure what `doc` is. Consider refactoring so before_create is passed a symbol with an intention revealing name, e.g.

```ruby
  before_create :generate_api_key # calls `generate_api_key` method
```